src/pytest_molecule/__init__.py

```diff
@@ -173,6 +173,19 @@ def __init__(self, name, parent):
             self.molecule_driver = data.get("driver", {}).get("name", "no_driver")
             self.add_marker(self.molecule_driver)
 
+            # check for known markers and add them
+            markers = data.get("markers", [])
+            if "xfail" in markers:
```

Comment:
Why do you need this check?

Reply:
I do not understand the question. `data` contains the `molecule.yml` data and we want to add the marker only if the `markers` section is present and if it contains one of them.